lib/external_data_sets.rb

```diff
@@ -0,0 +1,25 @@
+class VehicleFactory
+#attr_reader 
+def initialize
+  @factory_vehicles = []
+end
+
+def create_vehicles(api_data)
+  api_data.each do |vehicle_data|
+    car_data = {
+    vin: vehicle_data[:vin_1_10],
+    make: vehicle_data[:make],
+    model: vehicle_data[:model],
+    year: vehicle_data[:model_year],
+    engine: :ev,  
+    }
+    vehicle = Vehicle.new(car_data)
+    @factory_vehicles << vehicle
+  end
+  @factory_vehicles
+end
+
+#require 'pry';binding.pry
+
+end
+
```

Comment:
Your logic here looks good. One thing that I think you could remove is initializing your class with the `@factory_vehicles` empty array. If you removed that, you could utilize `map` in your `create_vehicles` method to return the transformed array. Something like: 

```ruby
def create_vehicles(api_data)
  api_data.map do |vehicle_data|
    car_data = {
    vin: vehicle_data[:vin_1_10],
    make: vehicle_data[:make],
    model: vehicle_data[:model],
    year: vehicle_data[:model_year],
    engine: :ev,  
    }
    vehicle = Vehicle.new(car_data)
  end
end
```